unlock-app/src/services/walletService.js

```diff
@@ -25,9 +22,10 @@ export const keyId = (lock, owner) => [lock, owner].join('-')
  * actually retrieving the data from the chain/smart contracts
  */
 export default class WalletService extends EventEmitter {
-  constructor(availableProviders = providers) {
+  constructor({ providers, unlockAddress }) {
     super()
-    this.providers = availableProviders
+    this.providers = providers
+    this.unlockAddress = unlockAddress
```

Comment:
Fun destructuring tip (not necessary to change, but just for the trivia):
```
Object.assign(this, { providers, unlockAddress })
```
or
```
// If you haven't already destructured the props (personally I think this one is kind of ugly)
({ providers: this.providers, unlockAddress: this.unlockAddress } = props)
```

Just in case you ever need to set a whole bunch of properties and want to save yourself some typing.